src/staking-v3/logic/services/DappStakingServiceEvm.ts

```diff
@@ -0,0 +1,244 @@
+import { inject, injectable } from 'inversify';
+import { IDappStakingService } from './IDappStakingService';
+import { DappStakingService } from './DappStakingService';
+import { DappStakeInfo } from '../models';
+import { IWalletService } from '../../../v2/services/IWalletService';
+import { IDappStakingRepository } from '../repositories';
+import { Symbols } from 'src/v2/symbols';
+import { evmPrecompiledContract } from 'src/modules/precompiled';
+import { IAccountUnificationRepository } from 'src/v2/repositories';
+import { Guard } from 'src/v2/common';
+
+const { dispatch } = evmPrecompiledContract;
+
+@injectable()
+export class DappStakingServiceEvm extends DappStakingService implements IDappStakingService {
+  private readonly wallet: IWalletService;
+
+  constructor(
+    @inject(Symbols.DappStakingRepositoryV3) dappStakingRepository: IDappStakingRepository,
+    @inject(Symbols.WalletFactory) walletFactory: () => IWalletService,
+    @inject(Symbols.AccountUnificationRepository),
+    private accountUnificationRepository: IAccountUnificationRepository
+  ) {
+    super(dappStakingRepository, walletFactory);
+    this.wallet = walletFactory();
+  }
+
+  // @inheritdoc
+  public async claimUnstakeAndUnlock(
+    contractAddress: string,
+    amount: number,
+    senderAddress: string,
+    successMessage: string
+  ): Promise<void> {
+    const ss58Address = await this.getSS58Address(senderAddress);
```

Comment:
`Guard.ThrowIfUndefined(senderAddress, 'senderAddress');` should go above this line

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/1102/commits/0494d397fc98f0913afaad8d7781d1525312800e) commit.